pkg/controllers/nodeclass/garbagecollection/controller.go

```diff
@@ -0,0 +1,185 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package garbagecollection
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/awslabs/operatorpkg/singleton"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+	nodeclaimutils "sigs.k8s.io/karpenter/pkg/utils/nodeclaim"
+
+	"github.com/mitchellh/hashstructure/v2"
+	"github.com/samber/lo"
+
+	v1 "github.com/aws/karpenter-provider-aws/pkg/apis/v1"
+	"github.com/aws/karpenter-provider-aws/pkg/operator/options"
+	"github.com/aws/karpenter-provider-aws/pkg/providers/instanceprofile"
+)
+
+const (
+	// GarbageCollectionDelay = 24 * time.Hour
+	GarbageCollectionDelay = 1 * time.Minute // Use 1 minute for testing purposes
+)
+
+type Controller struct {
+	kubeClient              client.Client
+	cloudProvider           cloudprovider.CloudProvider
+	instanceProfileProvider instanceprofile.Provider
+	region                  string
+}
+
+func NewController(kubeClient client.Client, cloudProvider cloudprovider.CloudProvider, instanceProfileProvider instanceprofile.Provider, region string) *Controller {
+	return &Controller{
+		kubeClient:              kubeClient,
+		cloudProvider:           cloudProvider,
+		instanceProfileProvider: instanceProfileProvider,
+		region:                  region,
+	}
+}
+
+func (c *Controller) getActiveProfiles(ctx context.Context) (map[string]struct{}, error) {
+	nodeClaims, err := nodeclaimutils.ListManaged(ctx, c.kubeClient, c.cloudProvider)
+	if err != nil {
+		return nil, fmt.Errorf("listing nodeclaims, %w", err)
+	}
+
+	activeProfiles := make(map[string]struct{})
```

Comment:
style: we prefer this over `make` / `new` in this project. There's no functional difference, it's just for consistency
```suggestion
	activeProfiles := map[string]struct{}{}
```

Reply:
Also, we have a set library that we use rather than using a `map[T]struct{}`